rosbag2_transport/src/rosbag2_transport/play_options.cpp

```diff
@@ -62,7 +62,9 @@ Node convert<rosbag2_transport::PlayOptions>::encode(
 bool convert<rosbag2_transport::PlayOptions>::decode(
   const Node & node, rosbag2_transport::PlayOptions & play_options)
 {
-  optional_assign<uint64_t>(node, "read_ahead_queue_size", play_options.read_ahead_queue_size);
+  optional_assign<decltype(play_options.read_ahead_queue_size)>(
+    node, "read_ahead_queue_size",
+    play_options.read_ahead_queue_size);
```

Comment:
this fixes the compile error, but it would be more straight-forward if we do,

```suggestion
  optional_assign<size_t>(node, "read_ahead_queue_size", play_options.read_ahead_queue_size);
```

or we can use `decltype` for else where?

```cpp
#define OPTIONAL_ASSIGN(node, fileld, assign_to) \
  do {
    optional_assign<decltype(assign_to)>(node, field, assign_to);
  } while (0)

OPTIONAL_ASSIGN(node, "read_ahead_queue_size", play_options.read_ahead_queue_size);
```

Reply:
@fujitatomoya Building with size_t was successful on macos, I accept your suggestion.

Reply:
I'm not familiar with yaml-cpp, so please let me know if this change breaks the current behavior.